L1Trigger/L1THGCal/plugins/HGCalTriggerGeometryESProducer.cc

```diff
@@ -39,23 +46,17 @@ HGCalTriggerGeometryESProducer::ReturnType HGCalTriggerGeometryESProducer::produ
   //using namespace edm::es;
   ReturnType geometry(HGCalTriggerGeometryFactory::get()->create(geometry_name_, geometry_config_));
   geometry->reset();
-  edm::ESHandle<CaloGeometry> calo_geometry;
-  iRecord.get(calo_geometry);
+  auto calo_geometry = iRecord.getHandle(calo_geometry_token_);
   // Initialize trigger geometry for V7/V8 HGCAL geometry
   if (calo_geometry.isValid() && calo_geometry->getSubdetectorGeometry(DetId::Forward, HGCEE) &&
       calo_geometry->getSubdetectorGeometry(DetId::Forward, HGCHEF) &&
       calo_geometry->getSubdetectorGeometry(DetId::Hcal, HcalEndcap)) {
-    geometry->initialize(calo_geometry);
+    geometry->initialize(calo_geometry.product());
   }
   // Initialize trigger geometry for V9 HGCAL geometry
   else {
-    edm::ESHandle<HGCalGeometry> ee_geometry;
-    edm::ESHandle<HGCalGeometry> hsi_geometry;
-    edm::ESHandle<HGCalGeometry> hsc_geometry;
-    iRecord.getRecord<IdealGeometryRecord>().get("HGCalEESensitive", ee_geometry);
-    iRecord.getRecord<IdealGeometryRecord>().get("HGCalHESiliconSensitive", hsi_geometry);
-    iRecord.getRecord<IdealGeometryRecord>().get("HGCalHEScintillatorSensitive", hsc_geometry);
-    geometry->initialize(ee_geometry, hsi_geometry, hsc_geometry);
+    geometry->initialize(
+        &iRecord.get(ee_geometry_token_), &iRecord.get(hsi_geometry_token_), &iRecord.get(hsc_geometry_token_));
```

Comment:
(repeating from the description) With the eventual (naive) ES-product prefetching, this implementation would always load/produce the `CaloGeometry` and the 3 `HGCalGeometry` objects. In practice (based on the comments above) the `HGCalGeometry` objects would be "extra" only in case of V7/V8 HGCal geometries. Are the V7/V8 geometries still relevant?

I presume the configurations for `HGCalGeometryESProducer` instances are always in the configuration (since that is the recommended pattern). Is it ok to run them for V7/V8 geometries? (just thinking if the price is only some CPU time and memory, or if something weird could happen as well)

There is a `phase2_hgcalV9` modifier. Would it be acceptable to use that to control the behavior of this ESProducer?

Reply:
I plan to get rid of the workflows with the old geometries to address #26938 (hopefully this week), and then the code support for them will be gradually removed (#27095). In general, using the modifier is good practice to make it clear when the relevant access method should be employed.

Reply:
>  this implementation would always load/produce the `CaloGeometry` and the 3 `HGCalGeometry` objects

Does it mean that the method
```cpp
bool isV9Geometry() const { return !calo_geometry_; }
```
would always return `false`?

Reply:
> > this implementation would always load/produce the `CaloGeometry` and the 3 `HGCalGeometry` objects
> 
> Does it mean that the method
> 
> ```c++
> bool isV9Geometry() const { return !calo_geometry_; }
> ```
> 
> would always return `false`?

No, because in case of V7/V8 geometry only the `calo_geometry_` member is set and thus the current behavior of `HGCalTriggerGeometryBase` is preserved. What I discussed in above was about how the production of the input ESProducts would change.

Reply:
@kpedro88 

> In general, using the modifier is good practice to make it clear when the relevant access method should be employed.

I take this as a green light for using the modifier here. Do I interpret the modifier/era structure correctly that it is sufficient to use only `phase2_hgcalV9`, or should it be `phase2_hgcalV9 | phase2_hgcalV10`?

Reply:
The workflows with HGCal V10 also include the V9 modifier to reduce duplication, so it's fine to use just the V9 one.